src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -0,0 +1,59 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import com.tinkerpop.blueprints.Graph
+import com.tinkerpop.blueprints.impls.orient.OrientConfigurableGraph.THREAD_MODE
+import com.tinkerpop.blueprints.impls.orient.{OrientGraph, OrientGraphFactory}
+
+import scala.util.{Failure, Success, Try}
+
+object DataSource {
+  def apply(url: String, user: String, password: String, minPoolSize: Int, maxPoolSize: Int) = {
+    val factory: OrientGraphFactory = createFactory(url, user, password)
+    factory.setupPool(minPoolSize, maxPoolSize)
+    new DataSource(factory)
+  }
+
+  def apply(url: String, user: String, password: String) = {
+    val factory: OrientGraphFactory = createFactory(url, user, password)
+    new DataSource(factory)
+  }
+
+  def createFactory(url: String, user: String, password: String): OrientGraphFactory = {
+    val factory = new OrientGraphFactory(url, user, password)
+    factory.setThreadMode(THREAD_MODE.MANUAL)
+    factory.setAutoStartTx(false)
+    factory
+  }
+}
+
+class DataSource(graphFactory: OrientGraphFactory) {
```

Comment:
do we need a test for success w/ commit and failure w/ rollback to verify each circumstance did to the db what we expected?  (maybe i missed seeing a test for this and it already exists)